### PR TITLE
GPT-2 tokenizer

### PR DESCRIPTION
**Pull request by dexterju**
**Patch description**
This will add a gpt2 stype bpe tokenizer inside Parlai
**Testing steps**
Added test in test_dict.py

### CODE REVIEW

**Comment by dexterju** [comment]
@stephenroller  I am not super sure about the test, https://github.com/facebookresearch/ParlAI/blob/master/tests/test_dict.py doesn't test any tokenizer which requires parameters config? Like nltk ones, where self.nltk will not be set unless this parameter is given at init time?

**Comment by dexterju** [comment]
Testing example: 
'Hello, ParlAI! 😀'
['Hello', ',', 'ĠPar', 'l', 'AI', '!', 'ĠðŁĺ', 'Ģ']

**Comment by stephenroller** [comment]
Does vec2text correctly reverse it?

File: parlai/core/dict.py [hunk 1]
```diff
@@ -447,6 +449,12 @@ def nltk_tokenize(self, text, building=False):
             for token in self.word_tok.tokenize(sent)
         )
 
+    def gpt2_bpe_tokenize(self, text):
```
Comment (stephenroller) [inline parlai/core/dict.py]: i think this has to be named `gpt2_tokenize`

**Comment by dexterju** [comment]
Should be good for review @stephenroller

File: parlai/core/gpt2_helper.py [hunk 2]
```diff
@@ -0,0 +1,161 @@
+#!/usr/bin/env python3
```
Comment (klshuster) [inline parlai/core/gpt2_helper.py]: should we consider a `utils` folder (perhaps not even in `core`)? it feels a little weird to me to have `gpt2_helper` in core

File: parlai/core/gpt2_helper.py [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+        self.byte_encoder = bytes_to_unicode()
+        self.byte_decoder = {v: k for k, v in self.byte_encoder.items()}
+        self.bpe_ranks = dict(zip(bpe_merges, range(len(bpe_merges))))
+        self.cache = {}
+        try:
+            import regex as re
+
+            self.re = re
+        except ImportError:
+            raise ImportError('Please install regex with: pip install regex')
```
Comment (klshuster) [inline parlai/core/gpt2_helper.py]: is this not a requirement in parlai already? seems like it should be if not

Comment (stephenroller) [reply parlai/core/gpt2_helper.py]: This was copy pasta from fairseq.

File: parlai/core/gpt2_helper.py (hunk 2 above)
Comment (stephenroller) [reply parlai/core/gpt2_helper.py]: I agree. We'll move things in another PR though.

**Comment by dexterju** [comment]
Merge : )

**Pull request merged**